### PR TITLE
haskell-ide-engine 0.1.0.0 (new formula)

### PR DESCRIPTION
**Pull request by lukel97**
Add a new formula for [haskell-ide-engine](https://github.com/haskell/haskell-ide-engine/issues/529).
Requires stack to build, and has an option to skip generating a Hoogle database since it can take quite a while.
Needs icu4c be explicitly installed via homebrew as the default macOS installed one is not sufficient for the text-icu package.

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/haskell-ide-engine.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class HaskellIdeEngine < Formula
+  desc "The IDE engine and LSP server for Haskell. Not an IDE"
+  homepage "https://github.com/haskell/haskell-ide-engine"
+
+  url "https://github.com/haskell/haskell-ide-engine.git",
+      :revision => "8e0b60f3d47f6332c0f4415b9fff03815e229275"
```
Comment (commitay) [inline Formula/haskell-ide-engine.rb]: We need a tagged stable release, not an arbitrary commit.

File: Formula/haskell-ide-engine.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  version "0.1.0.0"
+  sha256 "e2195fc6bf01e4b9509e7be388e53fb4e0103658a3f26aca997b54cdd52273dd"
```
Comment (commitay) [inline Formula/haskell-ide-engine.rb]: sha256 is redundant with revision

File: Formula/haskell-ide-engine.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  depends_on "haskell-stack" => :build
```
Comment (commitay) [inline Formula/haskell-ide-engine.rb]: Building with cabal is preferred and seems to be supported by upstream, is there a reason we can't use it?

Comment (ilovezfs) [reply Formula/haskell-ide-engine.rb]: Downloading binary versions of GHC during the build is not permissible, so we'll probably need to pass on this formula.

Comment (Bubba) [reply Formula/haskell-ide-engine.rb]: Is it be possible to provide a bottle/binary only version of this formula?

Comment (ilovezfs) [reply Formula/haskell-ide-engine.rb]: It should be built from source with cabal using
```
depends_on "ghc"
```
if it's going to be in Homebrew.

**Comment by Bubba** [comment]
@ilovezfs @commitay Is it acceptable to build formulae with `cabal new-build`?

**Comment by Bubba** [comment]
@ilovezfs @commitay this PR has been revised

**Comment by commitay** [comment]
ILZ has left the project so please don't ping them.

This should be changed to use `install_cabal_package`, e.g:
```ruby
require "language/haskell"

class HaskellIdeEngine < Formula
  include Language::Haskell::Cabal

  desc "The engine for haskell ide-integration. Not an IDE"
  homepage "https://github.com/haskell/haskell-ide-engine"
  url "https://github.com/haskell/haskell-ide-engine.git",
      :tag => "0.2.2.0",
      :revision => "f73e4698fabf44667f6bf3165d9fac8d091d2437"

  depends_on "cabal-install" => :build
  depends_on "ghc" => :build

  def install
    install_cabal_package
  end
```

**Comment by Bubba** [comment]
@commitay HIE cannot build with regular `cabal install` and needs to be built with `cabal new-build` instead, and from what I can tell `install_cabal_package` does the former

**Comment by Zialus** [comment]
Would it not be acceptable to build with stack?

**Comment by bubba** [comment]
Would it be preferable to add cabal new-build functionality into `Language::Haskell::Cabal` first?

**Comment by commitay** [comment]
Sorry, I'd forgotten about this PR.

> Would it be preferable to add cabal new-build functionality into `Language::Haskell::Cabal` first?

Yes, I've been thinking about if we should since `cabal-install` started printing warnings about legacy commands. I've been meaning to discuss it internally with the other maintainers but I haven't gotten around to it yet.

**Comment by fxcoudert** [comment]
Given lack of activity for a long time, closing for now.

**Pull request closed without merging**